### ISSUE TITLE
tiny gcode visualizer lib for tiny webserver

### ISSUE THREAD

**Issue by mkeyno** [patrickelectric/Gcode3D#2]
sorry to open ticket for my question but actually I'm looking for light gcode visualizer or probably cad2gcode convertor in my webpage , but this webpage hosted on tiny ESP826612F module which is connected to Arduino GRBL gcode interpreter by serial line, so I should have great care about its weight , My plan is just connect to wifi module and open the browser and do designated CAM job.,As I'm not pro in client side script like JavaScript I was hopping could have good advise here, do you suggest to make some plugin from such library so I shouldn't have to download or call all library ?
thanks

**Comment by patrickelectric** [comment]
@mkeyno Yes, I think that is possible. I have already saw html5 and js in esp8266, maybe doing some work over this examples is possible to integrate a minimalist 3d viewer like the one used in this project.

**Comment by patrickelectric** [comment]
BTW, if you would like to communicate with esp8266 via socket using a printer host, you can take a look in the project that is in development called [AtCore from KDE](https://github.com/KDE/atcore). We have a aim to do that in the future.

**Comment by mkeyno** [comment]
thanks @patrickelectric  actually the main problem  is   number of file dependency is lot and make webserver unstable and should consider, there is no internet connection and module is in access point mode, in this situation we gonna merge files in to the one, but still zipped file size is high and in many situation it is not fully loaded in client browser so I was thinking maybe it is good idea to make plugin from library so put the weight on browser and only call class instance in the program,
I might add I'm not good in JS and maybe not describe problem well
